harness/determined/searcher/search_runner.py

```diff
@@ -0,0 +1,328 @@
+import json
+import logging
+import os
+import pickle
+import time
+import uuid
+from pathlib import Path
+from typing import Any, Dict, List, Optional, Sequence, Tuple
+
+from determined.common.api import bindings
+from determined.common.api.bindings import (
+    determinedexperimentv1State,
+    v1SearcherEvent,
+    v1TrialExitedEarlyExitedReason,
+)
+from determined.experimental import client
+from determined.searcher.search_method import ExitedReason, Operation, Progress, SearchMethod
+
+EXPERIMENT_ID_FILE = "experiment_id"
```

Comment:
what do you think about `"experiment_id.txt"`?

I don't mind text files in unix without the .txt extension myself.  Just a thought.

Reply:
hmm, it actually happens to be a binary: